src/layouts/sidebar-sidecar/server.ts

```diff
@@ -110,8 +110,22 @@ export function getStaticGenerationFunctions<
         scope: await getScope(),
       })
 
+      /**
+       * Try to load the static props for the given context. If there is a
+       * ContentApiError with a 404 status, return a 404 status and page.
+       * https://nextjs.org/docs/api-reference/data-fetching/get-static-props#notfound
+       */
+      let loadStaticPropsResult
+      try {
+        loadStaticPropsResult = await loader.loadStaticProps(ctx)
+      } catch (error) {
+        if (error.status === 404) {
+          return { notFound: true }
+        }
```

Comment:
💭 🚧  Not sure, but I think we want to `throw` again here if the error is _not_ a 404.

```suggestion
        // Catch 404 errors, return a 404 status page
        if (error.status === 404) {
          return { notFound: true }
        }
        // We still want to throw non-404 errors
        throw error
```

Otherwise I think pages that have genuine 500 errors from calling `loader.loadStaticProps(ctx)` will end up swallowed, and we'll instead see errors later in this function since `loadStaticPropsResult` will still be undefined 🤔

Reply:
I was wondering about this as well! (noted in the "Anything else?" section)

Will update with your suggestion 😊